contracts/src/v0.8/workflow/dev/WorkflowRegistry.sol

```diff
@@ -0,0 +1,712 @@
+  /// - To get the workflowKey, use `computeHashKey` with the workflow owner's address and the workflow name, or
+  ///   perform an offchain equivalent of keccak256(abi.encodePacked(owner, workflowName)).
+  ///
+  /// This function performs the following steps:
+  /// - Validates the provided workflow metadata.
+  /// - Retrieves the workflow by the caller's address and `workflowName`.
+  /// - Updates only the fields that have changed.
+  /// - Ensures that the workflow ID (`newWorkflowID`) must change and at least one of the URLs must also change.
+  /// - Updates the `secretsURL` hash mappings if the `secretsURL` changes.
+  ///
+  /// Requirements:
+  /// - `binaryURL` must always be provided, as it is required.
+  /// - If only one field is being updated, the other fields must be provided with their current values to keep them unchanged, otherwise
+  ///   they will be treated as empty strings.
+  /// - The DON ID must be in the allowed list to perform updates.
+  /// - The caller must be an authorized address. This means that even if the caller is the owner of the workflow, if they were later
+  ///   removed from the authorized addresses list, they will not be able to perform updates.
+  ///
+  /// Emits:
+  /// - `WorkflowUpdatedV1` event indicating the workflow has been successfully updated.
+  ///
+  /// @param workflowKey The unique identifier for the workflow.
+  /// @param newWorkflowID The rehashed unique identifier for the workflow.
+  /// @param binaryURL The URL pointing to the WASM binary. Must always be provided.
+  /// @param configURL The URL pointing to the configuration file. Provide an empty string ("") to remove it.
+  /// @param secretsURL The URL pointing to the secrets file. Provide an empty string ("") to remove it.
+  function updateWorkflow(
+    bytes32 workflowKey,
+    bytes32 newWorkflowID,
+    string calldata binaryURL,
+    string calldata configURL,
+    string calldata secretsURL
+  ) external registryNotLocked {
+    _validateWorkflowMetadata(newWorkflowID, bytes(binaryURL).length, bytes(configURL).length, bytes(secretsURL).length);
+
+    WorkflowMetadata storage workflow = _getWorkflowFromStorage(msg.sender, workflowKey);
+
+    uint32 donID = workflow.donID;
+    _validatePermissions(donID, msg.sender);
+
+    // Store the old workflowID for event emission.
+    bytes32 currentWorkflowID = workflow.workflowID;
+
+    // Condition to revert: WorkflowID must change, and at least one URL must change
+    if (currentWorkflowID == newWorkflowID) {
+      revert WorkflowIDNotUpdated();
+    }
+
+    // Determine which URLs have changed
+    bool sameBinaryURL = Strings.equal(workflow.binaryURL, binaryURL);
+    bool sameConfigURL = Strings.equal(workflow.configURL, configURL);
+    bool sameSecretsURL = Strings.equal(workflow.secretsURL, secretsURL);
+    if (sameBinaryURL && sameConfigURL && sameSecretsURL) {
+      revert WorkflowContentNotUpdated();
```

Comment:
Hmm I don't think so in that one of the three must change. So it's possible for 2 of the 3 to remain unchanged but if no change is the desire the user would have to provide the same value as what is already in storage instead of leaving it out of the params